### PR TITLE
Healthbot rules for TCPIP

### PR DESCRIPTION
**Pull request by Surakshah**

### CODE REVIEW

**Comment by gvarao** [comment]
jenkins please retry a build

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 1]
```diff
@@ -0,0 +1,181 @@
+/*
+ * This rule checks for the neighbour ARP cache count on IRI(Internal routing interface).
+ *
+ * It shows yellow alarm when the count exceeds 90 percent of the maximum count.
+ * It shows red alarm when the count exceeds 95 percent of the maximum count or if there is an increase in drop count.
+ * It's sampling time is 1 minute.
+ */
+iceberg {
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Replace iceberg with healthbot

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    topic kernel.tcpip {
+        rule check-arp-cache-iri-count-iagent {
+            synopsis "Check arp iri cache count";
+            description "Check arp iri cache count";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Please make description more descriptive. e.g. This rule collects arp iri cache count and notify anomaly if arp iri cache count is not increasing.

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                        argument actual_val "$nbrcache-iri-cnt";
+                        argument percent_val "$arp-cache-iri-warning-percent";
+                    }
+                }
+                type integer;
+            }
+            trigger report-arp-iri-cache-count-iagent {
+                synopsis "Trigger to check arp iri cache count";
+                description "Trigger to check arp iri cache count";
+                frequency 1m;
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Always ensure trigger frequency is atleast 1.5 times of sensor frequency. Modify trigger frequency

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count.rule [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+ * This rule checks for the neighbour ARP cache count on IRI(Internal routing interface).
+ *
+ * It shows red alarm when the count exceeds 95 percent of the maximum count or if there is an increase in drop count.
+ * It shows yellow alarm when the count exceeds 90 percent of the maximum count.
+ * It's sampling time is 1 minute.
+ */
+healthbot {
+    topic kernel.tcpip {
+        rule check-arp-cache-iri-count {
+            synopsis "To check ARP IRI cache count in the system";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count.rule]: Modify as mentioned in previous rule

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+ *
+ * It shows yellow alarm when the count exceeds 90 percent of the maximum count.
+ * It shows red alarm when the count exceeds 95 percent of the maximum count or if there is an increase in drop count.
+ * It's sampling time is 1 minute.
+ */
+iceberg {
+    topic kernel.tcpip {
+        rule check-arp-cache-mgmt-count-iagent {
+            synopsis "Check arp management cache count";
+            description "Check arp management cache count";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Modify description as mentioned in previous rule

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+                        argument actual_val "$nbrcache-mgmt-cnt";
+                        argument percent_val "$arp-cache-mgmt-warning-percent";
+                    }
+                }
+                type integer;
+            }
+            trigger report-arp-mgmt-cache-count-iagent {
+                synopsis "Trigger to check arp management cache count";
+                description "Trigger to check arp management cache count";
+                frequency 1m;
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Always ensure trigger frequency is atleast 1.5 times of sensor frequency. Modify trigger frequency

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+                    }
+                }
+                term arp-mgmt-cache-count-critical {
+                    when {
+                        equal-to "$output-critical" 1;
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "arp management cache count is very high.arp management cache count:$nbrcache-mgmt-cnt Maximum arp management cache count: $nbrcache-mgmt-max ";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Start message with CAPITAL letter and remove white spaces

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+                    }
+                }
+                term arp-mgmt-cache-count-warning {
+                    when {
+                        equal-to "$output-warning" 1;
+                    }
+                    then {
+                        status {
+                            color yellow;
+                            message "arp management cache count is high.arp management cache count:$nbrcache-mgmt-cnt Maximum arp management cache count: $nbrcache-mgmt-max ";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Start message with CAPITAL letter and remove white spaces

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+                        }
+                    }
+                }
+                term default {
+                    then {
+                        status {
+                            color green;
+                            message "arp management cache count normal.arp management cache count:$nbrcache-mgmt-cnt Maximum arp management cache count: $nbrcache-mgmt-max  ";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Start message with CAPITAL letter and remove white spaces

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+                }
+            }
+            variable arp-cache-mgmt-critical-percent-input {
+                value 95;
+                description "arp management cache percentage to raise alarm when critical limit is reached";
+                type int;
+            }
+            variable arp-cache-mgmt-warning-percent-input {
+                value 90;
+                description "arp management cache percentage to raise alarm when warning limit is reached";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Start description with CAPITAL letter

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+                        status {
+                            color green;
+                            message "arp management cache count normal.arp management cache count:$nbrcache-mgmt-cnt Maximum arp management cache count: $nbrcache-mgmt-max  ";
+                        }
+                    }
+                }
+            }
+            variable arp-cache-mgmt-critical-percent-input {
+                value 95;
+                description "arp management cache percentage to raise alarm when critical limit is reached";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Start description with CAPITAL letter

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+                    file arp_iri_cache.yml;
+                    table checkArpIriCacheTable;
+                    frequency 1m;
+                }
+            }
+            field arp-cache-iri-critical-percent {
+                constant {
+                    value "{{arp-cache-iri-critical-percent-input}}";
+                }
+                description "arp iri cache percentage critical limit";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Start description with CAPITAL letter

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+            field output-critical {
+                formula {
+                    user-defined-function {
+                        function-name arp-cache-iri-status;
+                        argument max_val "$nbrcache-iri-max";
+                        argument actual_val "$nbrcache-iri-cnt";
+                        argument percent_val "$arp-cache-iri-critical-percent";
+                    }
+                }
+                type integer;
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Add description

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+            field output-warning {
+                formula {
+                    user-defined-function {
+                        function-name arp-cache-iri-status;
+                        argument max_val "$nbrcache-iri-max";
+                        argument actual_val "$nbrcache-iri-cnt";
+                        argument percent_val "$arp-cache-iri-warning-percent";
+                    }
+                }
+                type integer;
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Add description

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+                    when {
+                        not-equal-to "$nbrcache-iri-drop-cnt" 0;
+                        increasing-at-least-by-value "$nbrcache-iri-drop-cnt" {
+                            value 1;
+                        }
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "arp iri cache drops seen. Arp IRI cache drop count :$nbrcache-iri-drop-cnt ";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Start message with CAPITAL letter and remove white spaces

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+                    }
+                }
+                term arp-iri-cache-count-critical {
+                    when {
+                        equal-to "$output-critical" 1;
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "arp iri cache count is very high.arp iri cache count:$nbrcache-iri-cnt Maximum arp iri cache count: $nbrcache-iri-max ";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Start message with CAPITAL letter and remove white spaces

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+                            color yellow;
+                            message "arp iri cache count is high.arp iri cache count:$nbrcache-iri-cnt Maximum arp iri cache count: $nbrcache-iri-max ";
+                        }
+                    }
+                }
+                term default {
+                    then {
+                        status {
+                            color green;
+                            message "arp iri cache count normal.arp iri cache count:$nbrcache-iri-cnt Maximum arp iri cache count: $nbrcache-iri-max ";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Start message with CAPITAL letter and remove white spaces

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+                        }
+                    }
+                }
+            }
+            variable arp-cache-iri-critical-percent-input {
+                value 95;
+                description "arp iri cache percentage to raise alarm when critical limit is reached";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Start description with CAPITAL letter.

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+                type int;
+            }
+            variable arp-cache-iri-warning-percent-input {
+                value 90;
+                description "arp iri cache percentage to raise alarm when warning limit is reached";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Start description with CAPITAL letter.

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 20]
```diff
@@ -0,0 +1,181 @@
...
+                    file arp_mgmt_cache.yml;
+                    table checkArpMgmtCacheTable;
+                    frequency 1m;
+                }
+            }
+            field arp-cache-mgmt-critical-percent {
+                constant {
+                    value "{{arp-cache-mgmt-critical-percent-input}}";
+                }
+                description "arp management cache percentage critical limit";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Start description with CAPITAL letter.

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 21]
```diff
@@ -0,0 +1,181 @@
...
+            }
+            field arp-cache-mgmt-warning-percent {
+                constant {
+                    value "{{arp-cache-mgmt-warning-percent-input}}";
+                }
+                description "arp management cache percentage warning limit";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Start description with CAPITAL letter.

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 22]
```diff
@@ -0,0 +1,181 @@
...
+            }
+            field output-warning {
+                formula {
+                    user-defined-function {
+                        function-name arp-cache-mgmt-status;
+                        argument max_val "$nbrcache-mgmt-max";
+                        argument actual_val "$nbrcache-mgmt-cnt";
+                        argument percent_val "$arp-cache-mgmt-warning-percent";
+                    }
+                }
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Add description

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule [hunk 23]
```diff
@@ -0,0 +1,181 @@
...
+                    when {
+                        not-equal-to "$nbrcache-mgmt-drop-cnt" 0;
+                        increasing-at-least-by-value "$nbrcache-mgmt-drop-cnt" {
+                            value 1;
+                        }
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "arp management cache drops seen.arp management cache drop count: $nbrcache-mgmt-drop-cnt ";
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Start message with CAPITAL letter and remove white spaces

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule (hunk 8 above)
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Start message with CAPITAL letter and remove white spaces

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count.rule [hunk 24]
```diff
@@ -0,0 +1,180 @@
+/*
+ * This rule checks for the neighbour ARP cache count on management interface.
+ *
+ * It shows yellow alarm when the count exceeds 90 percent of the maximum count.
+ * It shows red alarm when the count exceeds 95 percent of the maximum count or if there is an increase in drop count.
+ * It's sampling time is 1 minute.
+ */
+iceberg {
+    topic kernel.tcpip {
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count.rule]: Please take care of all above review comments in below rules

File: community_supplied/Kernel/kernel-tcpip/arp-iri-cache-count-analyser.playbook [hunk 25]
```diff
@@ -0,0 +1,5 @@
+iceberg {
```
Comment (gvarao) [inline community_supplied/Kernel/kernel-tcpip/arp-iri-cache-count-analyser.playbook]: relace "iceberg" with "healthbot" in all rule and playbook files as configuration object name changed.

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule (hunk 1 above)
Comment (Surakshah) [reply community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Addresses

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule (hunk 3 above)
Comment (Surakshah) [reply community_supplied/Kernel/kernel-tcpip/check-arp-cache-iri-count-iagent.rule]: Addressed

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule (hunk 5 above)
Comment (Surakshah) [reply community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count-iagent.rule]: Addressed

File: community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count.rule (hunk 24 above)
Comment (Surakshah) [reply community_supplied/Kernel/kernel-tcpip/check-arp-cache-mgmt-count.rule]: Modified icerberg to healthbot

**Comment by gvarao** [comment]
jenkins please retry a build

**Pull request merged**